server/models/comment.js

```diff
@@ -86,16 +97,31 @@ class Comment extends Model {
       ])
       .where('creator_id', this.creatorId);
 
-    const trigger = await Trigger.query().where({ name: 'comment_create' }).returning('id');
+    const badgesIds = await knex('user_badges').pluck('badge_id');
 
-    if (results[0].totalcomments > 1) {
-      const badges = await Badges.query().where('trigger_id', trigger[0].id);
-      await UserBadges.query().insert({ 'user_id': this.creatorId, 'badge_id': badges[0].id }).returning(['user_id']);
-    }
+    console.log(results[0].totalreplies, results[0].totalcomments, badgesIds);
+    try {
+      const comment_create = await Trigger.query().where({ name: 'comment_create' }).returning('id');
+      const comment_create_badges = await Badges.query().where('trigger_id', comment_create[0].id);
 
-    if (results[0].totalreplies > 3) {
-      const badges = await Badges.query().where('trigger_id', trigger[0].id);
-      await UserBadges.query().insert({ 'user_id': this.creatorId, 'badge_id': badges[0].id }).returning(['user_id']);
+      for (const badge of comment_create_badges) {
+        if (!badgesIds.includes(badge.id) & results[0].totalcomments > badge.frequency & results[0].totalcomments > 0) {
```

Comment:
I believe the check should be when total comments equals the frequency.
Also, you are doing bitwise operation?

Reply:
That makes more sense, included it in the review

Reply:
include a check so that users can only unlock published badges.. a badge can also expire in which case no one should be able to earn it.